From 81aeff64863ceb6abcae607b07de8794f85d2014 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 4 May 2023 16:47:39 +0200
Subject: [PATCH 9/9] Ensure sshd service is enabled

Make sure that we actually start sshd during early phase
---
 lib/tiny-cloud/init   |  7 +++++++
 tests/init-early.test | 12 +++++++++++-
 2 files changed, 18 insertions(+), 1 deletion(-)

diff --git a/lib/tiny-cloud/init b/lib/tiny-cloud/init
index 42931bb..1e5f828 100644
--- a/lib/tiny-cloud/init
+++ b/lib/tiny-cloud/init
@@ -15,6 +15,7 @@ INIT_ACTIONS_EARLY="
     install_hotplugs
     set_network_config
     create_default_user
+    enable_sshd
 "
 INIT_ACTIONS_MAIN="
     save_userdata
@@ -101,6 +102,12 @@ init__create_default_user() {
     echo 'permit nopass :wheel' > "$ROOT/etc/doas.d/wheel.conf"
 }
 
+init__enable_sshd() {
+    $MOCK rc-update add sshd default
+    # incase something else has enabled/disabled dservices
+    $MOCK rc-update --update
+}
+
 ### init-main functions
 
 init__set_hostname() {
diff --git a/tests/init-early.test b/tests/init-early.test
index d640ace..a97cfd2 100755
--- a/tests/init-early.test
+++ b/tests/init-early.test
@@ -9,7 +9,8 @@ lib="$srcdir"/lib/tiny-cloud/init
 init_tests \
 	expand_root \
 	expand_root_partition \
-	install_hotplugs_fail
+	install_hotplugs_fail \
+	enable_sshd
 
 PROVIDERS="aws azure gcp nocloud oci"
 
@@ -46,3 +47,12 @@ install_hotplugs_fail_body() {
 		-e match:"vnic_eth_hotplug\(!\)" \
 		sh -c ". $lib; HOTPLUG_MODULES='vnic_eth_hotplug'; init__install_hotplugs"
 }
+
+enable_sshd_body() {
+	for provider in $PROVIDERS; do
+		CLOUD="$provider" atf_check \
+			-o match:"rc-update.* add sshd default" \
+			-o match:"rc-update.* --update" \
+			sh -c ". $lib; init__enable_sshd"
+	done
+}
-- 
2.40.1

